Allows comm consoles to handle arbitrary evacuation controller types

thasc opened the pull request:
This change removes the command and communication console's explicit understanding of evacuation. Instead, it calls `evacuation_controller.available_evac_options()` to obtain a list of available `evacuation_options` for the current situation, renders them naively into the UI, and allows the selected option to decide what happens next via `evacuation_option.execute()`.

This also gently updates the styling on the command and communication program, and adds an option for initiating a bluespace jump on Torch's pods-type controller. It also adds the possibility to add an additional delay to crew transfer or emergency evac for any given evac controller (to enable Torch jump prep to take longer than abandoning ship).

Fixes #15163, long story short.

_Hurt me good, peer review, daddy likes it_

thasc committed:
Comm consoles can now deal with arbitrary evacuation controllers.

Evacuation controllers now have two new functions - available_evac_options and handle_evac_option - to support this indirection.

Mild spruce up of command and communications program.

thasc committed:
Tweaks and improvements to evacuation abstraction. Also adds bluespace jump evac option to evacuation_pods.

thasc committed:
Adds option to include an additional delay to emergency evacuation or crew transfer independently (e.g. if crew transfer should be a more relaxed affair).

thasc commented on code/controllers/evacuation/evacuation_pods.dm:
```diff
@@ -24,3 +38,85 @@
+	needs_syscontrol = TRUE
+	silicon_allowed = TRUE
+
+/datum/evacuation_option/abandon_ship/execute(mob/user)
```
There's some code duplication going on between the `execute` functions that could stand to be gone. I'm too tired to think of how best to do it. Answers on a postcard please

comma replied:
Not much prettier, but can move this code up in base class, and define messages as vars, like using_map.emergency_shuttle_leaving_dock etc do

thasc replied:
The best solution I came up with will take a fair bit of work and partly re-architect the evacuation controller (highly generic evacuation controller that passes calls on to one of multiple 'evacuation methods'), so I've undone the partly misleading code de-duplication I tried and left this as is for now.

thasc commented:
I need to ponder through this a bit more - it works, but there's a couple of little hangnails - but hopefully should be sorted tonight.

thasc committed:
Bit more explicit state machine usage in the pods controller.

No options will be made available on the pods controller if the evac controller is on cooldown.

thasc commented:
I decided the way I was handling code duplication sucked and undid it. I have a GREATER SCHEME in mind to resolve it more thoroughly but this works for now and the GREATER SCHEME can wait for a different PR that isn't a bugfix.
